mkosi/__init__.py

```diff
@@ -2636,6 +2677,11 @@ def unlink_output(config: MkosiConfig) -> None:
             unlink_try_hard(config.output_split_kernel_image)
             unlink_try_hard(config.output_split_initrd)
             unlink_try_hard(config.output_split_cmdline)
+            unlink_try_hard(config.output_split_kmods_initrd)
+            i = 0
+            while config.output_split_initrds(i).exists():
+                unlink_try_hard(config.output_split_initrds(i))
+                i += 1
```

Comment:
Jugling indices is ugly and checking whether something exists before removing it is inherently racy. The problem is, that there's no way get all possible initrds without iterating over them until one doesn't exist (what if one in the middle is missing for some reason?), because then one could just `unlink_try_hard` each of them, since that's a noop if the file doesn't exist (anymore).

Reply:
Maybe I could use [`remove_glob`](https://github.com/systemd/mkosi/blob/aaaed2a5e409efbb1ddf3b87be14a92c08b08cc3/mkosi/__init__.py#L2591) here.